File: fitness-checkin-backend/src/main/java/com/fitness/checkin/service/UserService.java
package com.fitness.checkin.service;

import com.fitness.checkin.entity.User;

/**
 * 用户服务接口
 * 提供用户相关的业务逻辑
 * 
 * @author Kou
 * @version 1.0.0
 */
public interface UserService {

    /**
     * 根据用户ID获取用户信息
     * 
     * @param userId 用户ID
     * @return 用户实体
     */
    User getUserById(Long userId);

    /**
     * 根据openid获取用户信息
     * 
     * @param openid 微信openid
     * @return 用户实体
     */
    User getUserByOpenid(String openid);

    /**
     * 创建或获取用户
     * 如果用户不存在则创建新用户，否则返回现有用户
     * 
     * @param openid    微信openid
     * @param nickname  昵称
     * @param avatarUrl 头像URL
     * @return 用户实体
     */
    User createOrGetUser(String openid, String nickname, String avatarUrl);

    /**
     * 更新用户信息
     * 
     * @param userId    用户ID
     * @param nickname  昵称
     * @param avatarUrl 头像URL
     * @return 更新后的用户实体
     */
    User updateUser(Long userId, String nickname, String avatarUrl);

    /**
     * 检查用户是否存在
     * 
     * @param openid 微信openid
     * @return 是否存在
     */
    boolean existsByOpenid(String openid);
}

File: src/services/UserService.ts
import { request } from '../utils/request'
import { User, LoginResult, UpdateUserInfoRequest, APIResponse } from '../types'

/**
 * 用户服务类
 * 处理用户相关的API调用
 */
export class UserService {
  /**
   * 用户登录
   * 后端 AuthController.wxLogin 返回扁平 Map {token, userId, openid, nickname, avatarUrl}
   * @param loginData 登录数据
   * @returns 登录结果（扁平结构）
   */
  static async login(loginData: {
    code: string
    nickname?: string
    avatarUrl?: string
    gender?: number
    province?: string
    city?: string
    country?: string
  }): Promise<APIResponse<LoginResult>> {
    return request('/auth/login', 'POST', loginData)
  }

  /**
   * 获取当前用户信息
   * @returns 用户信息
   */
  static async getUserInfo(): Promise<APIResponse<User>> {
    return request('/auth/userinfo')
  }

  /**
   * 更新用户信息（部分字段：nickname / avatarUrl 至少一个；avatarUrl 空串清空头像）
   * @param userData 用户数据
   * @returns 更新后的用户信息
   */
  static async updateUserInfo(userData: UpdateUserInfoRequest): Promise<APIResponse<User>> {
    return request('/auth/userinfo', 'PUT', userData)
  }

  /**
   * 根据用户ID获取用户信息
   * @param userId 用户ID
   * @returns 用户信息
   */
  static async getUserById(userId: string): Promise<APIResponse<User>> {
    return request(`/users/${userId}`)
  }

  /**
   * 批量获取用户信息
   * @param userIds 用户ID列表
   * @returns 用户信息列表
   */
  static async getUsersByIds(userIds: string[]): Promise<APIResponse<User[]>> {
    return request('/users/batch', 'POST', { userIds })
  }
}
